File: frontend/src/app/features/caisse/models/depense-caisse-rattachement-paie-request.ts
import { TypePaiementPersonnel } from './type-paiement-personnel';

export interface DepenseCaisseRattachementPaieRequest {
  employeId: number;
  periodePaie: string;
  typePaiementPersonnel: TypePaiementPersonnel;
  motif?: string;
  motifRetenue?: string;
  motifPaiementPartiel?: string;
  retenueDefinitive?: boolean;
  commentaireCorrection: string;
}

File: backend/src/main/java/com/mini/credit/dto/caisse/DepenseCaisseRattachementPaieRequest.java
package com.mini.credit.dto.caisse;

import com.mini.credit.enums.TypePaiementPersonnel;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class DepenseCaisseRattachementPaieRequest {

    @NotNull
    private Long employeId;

    @NotBlank
    private String periodePaie;

    @NotNull
    private TypePaiementPersonnel typePaiementPersonnel;

    private String motif;
    private String motifRetenue;
    private String motifPaiementPartiel;
    private Boolean retenueDefinitive;

    @NotBlank
    private String commentaireCorrection;
}